core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigInstantiator.java

```diff
@@ -86,19 +89,37 @@ private static void handleObject(String prefix, Object o, SmallRyeConfig config)
                     } else if (name.equals(ConfigItem.ELEMENT_NAME)) {
                         name = field.getName();
                     }
-                    String fullName = prefix + "." + name;
+
                     final Type genericType = field.getGenericType();
                     final Converter<?> conv = getConverterFor(genericType);
-                    try {
-                        Optional<?> value = config.getOptionalValue(fullName, conv);
-                        if (value.isPresent()) {
-                            field.set(o, value.get());
-                        } else if (!configItem.defaultValue().equals(ConfigItem.NO_DEFAULT)) {
-                            //the runtime config source handles default automatically
-                            //however this may not have actually been installed depending on where the failure occured
-                            field.set(o, conv.convert(configItem.defaultValue()));
+
```

Comment:
This only adds support for property renaming at build time, not at run time.

It would be better to implement this by modifying the data model to add the old names, and then use a configuration interceptor to implement the name redirection instead of digging into this already complex bit of code.  Or wait for smallrye/smallrye-config#310 to be complete and implement the feature there.